process/best_practices_badge.md

```diff
@@ -0,0 +1,57 @@
+- [Secure Software Development: Verification and More Specialized Topics (LFD106x) - Linux Foundation - Training](https://training.linuxfoundation.org/training/secure-software-development-verification-and-more-specialized-topics-lfd106/) 
+
+## Is there a list or document which describes the static analysis tools that ASWF supports or recommends?
+
+SonarCloud is the tool that multiple projects have implemented, and there is already an organization-level access token that projects can use with it. You can see a specific example in the OpenEXR analysis_workflow.yml, which shows how to build, run the test suite, post-process, and upload to SonarCloud using the SONAR_TOKEN from the organization level. You can see the results for the various ASWF projects with SonarCloud here:
+
+[https://sonarcloud.io/organizations/academysoftwarefoundation/projects](https://www.google.com/url?q=https://sonarcloud.io/organizations/academysoftwarefoundation/projects&sa=D&source=docs&ust=1675375752427760&usg=AOvVaw3aJ1DsMv9yF2Xf-Uhv8EKo)
+
+I'm also seeing some work towards using clang-tidy in Imath: [https://cs.github.com/AcademySoftwareFoundation/Imath/blob/3ad5d4da02795dbf37b44c5bbd2f919a7c01a19a/config/ImathSetup.cmake#L118](https://www.google.com/url?q=https://cs.github.com/AcademySoftwareFoundation/Imath/blob/3ad5d4da02795dbf37b44c5bbd2f919a7c01a19a/config/ImathSetup.cmake%23L118&sa=D&source=docs&ust=1675375752432129&usg=AOvVaw1HThXkwYas6U7fU6_IKetY)
+
+and in OSL:
+
+[https://cs.github.com/AcademySoftwareFoundation/Imath/blob/3ad5d4da02795dbf37b44c5bbd2f919a7c01a19a/config/ImathSetup.cmake#L118](https://www.google.com/url?q=https://cs.github.com/AcademySoftwareFoundation/Imath/blob/3ad5d4da02795dbf37b44c5bbd2f919a7c01a19a/config/ImathSetup.cmake%23L118&sa=D&source=docs&ust=1675375752432249&usg=AOvVaw10Qy3WPYpoaDl6XTJfp3F9)
+
+## "The project MUST have a documented process for responding to vulnerability reports." - does GitHub have anything to help with this?
+
+
+
+## "The project MUST list external dependencies in a computer-processable way." - is there an automated way to build this?
+
+LFX Security automatically will pull the list of dependencies for a project provided they use the standard mechanisms leveraged by Synk.
+
+## "The project MUST provide an assurance case that justifies why its security requirements are met. The assurance case MUST include: a description of the threat model, clear identification of trust boundaries, an argument that secure design principles have been applied, and an argument that common implementation security weaknesses have been countered."
+
+
+
+## The project MUST have a [reproducible build](https://reproducible-builds.org/). If no building occurs (e.g., scripting languages where the source code is used directly instead of being compiled), select "not applicable"
+
+The easiest solution is not to provide a binary build of the project, which is generally recommended. 
```

Comment:
Worth noting that this conflicts with the constant requests we get (especially for projects that are tricky to build) to provide binaries, for example in PyPI.

Reply:
We've also talked about taking a less draconian view of reproducible builds (i.e. not necessarily bit for bit identical). This is probably not a bad place to plug the aswf-docker containers, since they provide as reproducible a build environment as you can reasonably easily get to.

Reply:
>(i.e. not necessarily bit for bit identical).
Reproducable builds are bit for bit. There is two aspects (or even more) to reproducable builds: 1) Building on the same machine twice in a row produces the same binaries (called reproducable build system, https://reproducible-builds.org/docs/deterministic-build-systems/), 2) building on two different machines produce the same binaries (called reproducable build environment: https://reproducible-builds.org/docs/perimeter/). 1 is doable and not too hard. There is variances to 1, like building on the same machine but in different paths, changing the timezone, etc.

Reply:
One benefit of building project binaries through GitHub Actions is that users can easily reproduce the same results in their own fork of the project, and the two generated packages ought to be bit-identical.

As @jfpanisset suggests, ASWF Docker containers can also be used in this role, though for the MaterialX project we use the off-the-shelf virtual environments in GitHub Actions, and upload the binaries generated by this process for users to access.

Reply:
>One benefit of building project binaries through GitHub Actions is that users can easily reproduce the same results in their own fork of the project, and the two generated packages ought to be bit-identical.

Note though that the VM images used in G Actions cannot be relied on for reproducible builds. GH do slow roll-outs of images, which means one repo can use an image X while another (a fork for example) use Y and there is no way to tell GH Actions to use a specific image version, apart from saying `ubuntu-22.04`, but that's just a label, the real VM image version cannot be pinned.

Also docker is only an option on Linux and Windows. macOS doesn't have a concept of containers (at least not like Linux and Windows). But docker can be used on macOS (though I'm not sure if it's possible in GH Actions), which means cross compilation could be used (if possible, because it won't always be possible or might be hard to do on some projects).

Reply:
That's a good point, @JeanChristopheMorinPerso, and I wonder if this ongoing evolution of the virtual environments in GitHub Actions would break the notion of "reproducible builds" in the sense that the OpenSSF intends.  They're reproducible in the sense of being deterministic for any given virtual environment snapshot, but they will indeed change when the virtual environments themselves change.

Reply:
Good discussion - is there something we should add in the doc to capture this?

Reply:
Maybe we could simply say that more investigation is required? As far as I know we didn't try to have reproducible builds on any of our projects. I think that once we do it on one or two projects (say one C++ project and one Python project) we'll have a good idea of what we can expect from all our projects?